apps/aehttp/src/sc_ws_handler.erl

```diff
@@ -559,7 +578,17 @@ prepare_handler(Params) ->
         fun(_, {error, _} = Err) -> Err;
             (Fun, Accum) -> Fun(Accum)
         end,
-        #handler{}, Validators).
+        #handler{protocol = protocol(Params)}, Validators).
+
+protocol(#{protocol := P}) ->
+    case P of
+        <<"legacy">>  -> legacy;
+        <<"jsonrpc">> -> jsonrpc;
```

Comment:
No harm, but binary `protocol` should never happen?

Reply:
perhaps...